src/ddmd/intrange.d

```diff
@@ -77,35 +78,70 @@ struct SignExtendedNumber
             return 0;
     }
 
-    SignExtendedNumber opNeg() const
+    private alias SignExtendedNumber R;
+
+    R opUnary(string op : "++")()
+    {
+        if (value != UINT64_MAX)
+            ++value;
+        else if (negative)
+        {
+            value = 0;
+            negative = false;
+        }
+        return this;
+    }
+
+    R opUnary(string op : "~")() const
+    {
+        return SignExtendedNumber(~value);
+    }
+
+    R opUnary(string op : "-")() const
     {
         if (value == 0)
             return SignExtendedNumber(-cast(ulong)negative);
         else
             return SignExtendedNumber(-value, !negative);
     }
 
-    SignExtendedNumber opAdd(const SignExtendedNumber a) const
+    R opBinary(string op : "&")(R rhs) const
     {
-        uinteger_t sum = value + a.value;
-        bool carry = sum < value && sum < a.value;
-        if (negative != a.negative)
+        return SignExtendedNumber(value & rhs.value);
```

Comment:
same question here about signedness of result - looks like any two inputs produce a positive number